src/arc_spice/eval/inference_utils.py

```diff
@@ -0,0 +1,133 @@
+from collections.abc import Callable
+
+import torch
+from sklearn.metrics import hamming_loss, zero_one_loss
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+
+from arc_spice.data.multieurlex_utils import MultiHot
+from arc_spice.eval.translation_error import get_comet_model
+from arc_spice.variational_pipelines.RTC_single_component_pipeline import (
+    RTCSingleComponentPipeline,
+)
+from arc_spice.variational_pipelines.RTC_variational_pipeline import (
+    RTCVariationalPipeline,
+)
+
+
+class ResultsGetter:
+    def __init__(self, n_classes):
+        self.func_map: dict[str, Callable] = {
+            "recognition": self.recognition_results,
+            "translation": self.translation_results,
+            "classification": self.classification_results,
+        }
+        self.comet_model = get_comet_model()
+        self.multihot = MultiHot(n_classes)
+
+    def get_results(
+        self,
+        clean_output: dict[str, dict],
+        var_output: dict[str, dict],
+        test_row: dict[str, list[int]],
+        results_dict: dict[str, dict[str, list]],
+    ):
+        for step_name in clean_output:
+            results_dict = self.func_map[step_name](
+                test_row, clean_output, var_output, results_dict
+            )
+        results_dict["input_data"]["celex_ids"].append(test_row["celex_id"])
+        return results_dict
+
+    def recognition_results(self, test_row, clean_output, var_output, results_dict):
+        assert test_row is not None
+        assert clean_output is not None
+        assert var_output is not None
+        # ### RECOGNITION ###
+        # TODO: add this into results_getter
+        return results_dict
+
+    def translation_results(self, test_row, clean_output, var_output, results_dict):
+        # ### TRANSLATION ###
+        source_text = test_row["target_text"]
+        target_text = test_row["target_text"]
+        clean_translation = clean_output["translation"]["full_output"]
+
+        # load error model
```

Comment:
does this load the model? it looks already loaded

Reply:
I can update these comments, I think it used to be loaded here maybe, and I shifted it to the `__init__`.